src/services/lnd/schema.ts

```diff
@@ -46,15 +46,24 @@ const pathSchema = new Schema({
 })
 
 const paymentSchema = new Schema<LnPaymentType>({
-  status: { type: String, enum: ["settled", "failed", "pending"], required: true },
+  status: {
+    type: String,
+    enum: ["settled", "failed", "pending"],
+    required: true,
+  },
   confirmedAt: Date,
   createdAt: Date,
   destination: String,
   milliSatsFee: {
     type: Number,
     default: 0,
   },
-  paymentHash: String,
+  paymentHash: {
+    type: String,
+    index: true,
+    unique: true,
```

Comment:
is this unique? what happen when there are different payment attempt for the same hash?

Reply:
Based on what Alex said, these attempts are stored separately internally to `lnd` but are only ever exposed as a single object with latest state when accessed via the api. He was saying too that they had some sort of an attempts arrays which I was experimenting with a few weeks back but couldn't find.

Decided to make this unique because either way it's only one object that would ever get returned from `lnd` based on what they said and we've observed so far.

Reply:
so this means that we only get the last payment from the api? not the history?

Reply:
As far as I know so far yes

Reply:
Some feedback from Alex on this: https://github.com/alexbosworth/lightning/issues/50#issuecomment-965833080